pkg/tekton/metapipeline/metapipeline.go

```diff
@@ -0,0 +1,233 @@
+// to modify the pipeline (via modifying the configuration on the file system) and finally triggering the actual
+// pipeline build.
+// An error is returned in case the creation of the Tekton CRDs fails.
+func CreateMetaPipelineCRDs(params CRDCreationParameters) (*tekton.CRDWrapper, error) {
+	parsedPipeline, err := createPipeline(params)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO verify 'nil' for taskParams
+	pipeline, tasks, structure, err := parsedPipeline.GenerateCRDs(params.PipelineName, params.BuildNumber, params.Namespace, params.PodTemplates, nil, params.SourceDir)
+	if err != nil {
+		return nil, err
+	}
+
+	pipelineResourceName := tekton.PipelineResourceName(params.GitInfo, params.Branch, params.Context)
+	resources := []*pipelineapi.PipelineResource{tekton.GenerateSourceRepoResource(pipelineResourceName, params.GitInfo, params.Revision)}
+	// TODO verify 'nil' for pipelineParams
+	run := tekton.CreatePipelineRun(resources, pipeline.Name, pipeline.APIVersion, params.Labels, params.Trigger, params.ServiceAccount, nil)
+
+	tektonCRDs, err := tekton.NewCRDWrapper(pipeline, tasks, resources, structure, run)
+	if err != nil {
+		return nil, err
+	}
+
+	return tektonCRDs, nil
+}
+
+// createPipeline builds the parsed/typed pipeline which servers as input for the Tekton CRD creation.
+func createPipeline(params CRDCreationParameters) (*syntax.ParsedPipeline, error) {
+	steps, err := BuildExtensionSteps(params.Apps)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create app extending pipeline steps")
+	}
+
+	stage := syntax.Stage{
+		Name:  appExtensionStageName,
+		Steps: steps,
+		Agent: &syntax.Agent{
+			// TODO Agent does not seem to be used anywhere
+			Image: "unused",
+		},
+	}
+
+	parsedPipeline := &syntax.ParsedPipeline{
+		Stages: []syntax.Stage{stage},
+	}
+
+	env := buildEnvParams(params)
+	parsedPipeline.AddContainerEnvVarsToPipeline(env)
+
+	return parsedPipeline, nil
+}
+
+// GetExtendingApps returns the lst of apps which are installed in the cluster registered for extending the pipeline.
+// An app registers its interest in extending the pipeline by having the 'pipeline-extension' label set.
+func GetExtendingApps(jxClient versioned.Interface, namespace string) ([]jenkinsv1.App, error) {
+	listOptions := metav1.ListOptions{}
+	listOptions.LabelSelector = fmt.Sprintf(apps.AppTypeLabel+" in (%s)", apps.PipelineExtension)
+	appsList, err := jxClient.JenkinsV1().Apps(namespace).List(listOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "error  retrieving pipeline contributor apps")
+	}
+	return appsList.Items, nil
+}
+
+// BuildExtensionSteps builds a step (container) for each of the apps specified.
+func BuildExtensionSteps(extendingApps []jenkinsv1.App) ([]syntax.Step, error) {
+	var steps []syntax.Step
+
+	// TODO - Add step to create effective pipeline. This way apps will always have a pipeline config to load and work with,
```

Comment:
@abayer, so here I'd like to create the step which generates the effective jenkins-x.yml. Maybe extract that into a new `step syntax command`

Reply:
Yeah, I'm gonna do that in a separate PR today or tomorrow.